src/rust/en.nepnep/src/lib.rs

```diff
@@ -157,9 +148,46 @@ fn get_manga_list(filters: Vec<Filter>, page: i32) -> Result<MangaPageResult> {
 	})
 }
 
+pub fn initialize_hot_update() {
+	init_page("/hot.php", "vm.HotUpdateJSON = ", unsafe {
+		&mut HOT_UPDATE_RID
+	})
+}
+
 #[get_manga_listing]
-fn get_manga_listing(_listing: Listing, _page: i32) -> Result<MangaPageResult> {
-	todo!()
+fn get_manga_listing(listing: Listing, page: i32) -> Result<MangaPageResult> {
+	let listing_name = listing.name.as_str();
+	match listing_name {
+		"Hot Updates" => {
+			if unsafe { HOT_UPDATE_RID } < 0 {
+				initialize_hot_update()
+			}
+		}
+		_ => {
+			println!("Received unexpected listing: {}", listing_name);
+			return Err(aidoku::error::AidokuError {
+				reason: aidoku::error::AidokuErrorKind::Unimplemented,
+			});
```

Comment:
honestly probably better to do this instead but it's a pretty minor issue. AidokuError constructing is kinda ugly and it's not really a relevant error.
```suggestion
			panic!("Received unexpected listing: {}", listing_name);
```